pkg/cosign/git/github/github.go

```diff
@@ -145,6 +151,20 @@ func (g *Gh) PutSecret(ctx context.Context, ref string, pf cosign.PassFunc) erro
 	return nil
 }
 
+func (g *Gh) getPublicKey(ctx context.Context, client *github.Client, owner string, repo string) (*github.PublicKey, *github.Response, error) {
```

Comment:
sorry, I would like to understand why you are doing the `(g *Gh).` What is the benefit of doing that and not using a internal pure function?

```suggestion
func  getPublicKey(ctx context.Context, client *github.Client, owner string, repo string) (*github.PublicKey, *github.Response, error) {
```


same for the other one

Reply:
if the `g` had the github client that makes more sense, but it does not have, and you are passing as paramenter, i think you can drop the g thing

Reply:
no reason at all, I converted the methods to functions... initially I duplicated the original function and they I cleaned the code to avoid duplicated code and I ended up by leaving this as a method

Reply:
functions are ok, but that does not need to be part of `(g *Gh)` please drop that